File: backend/src/main/kotlin/ru/ivakin/mtgcollection/model/Card.kt
package ru.ivakin.mtgcollection.model;

import org.springframework.data.relational.core.mapping.Table
import org.springframework.data.annotation.Id

@Table("card")
data class Card(
    @Id val id: Long,
    val name: String,
    val setCode: String,
    val type: String?,
    var tags: List<String>?,
    var abilities: List<String>?,
    val power: Int?,
    val toughness: Int?,
    val manaNeutral: Int?,
    val manaWhite: Int?,
    val manaBlack: Int?,
    val manaGreen: Int?,
    val manaBlue: Int?,
    val manaRed: Int?,
    val description: String?,
    val commentary: String?,
    val approved: Int?
)

File: frontend/src/common/classes/Card.ts
export default class Card {
  id: number;

  name = '';

  setCode = '';

  type = '';

  tags = [''];

  abilities = [''];

  power = 0;

  toughness = 0;

  manaNeutral = 0;

  manaWhite = 0;

  manaBlack = 0;

  manaGreen = 0;

  manaBlue = 0;

  manaRed = 0;

  description = '';

  approved = false;

  commentary = '';

  constructor(
    id: number,
  ) {
    this.id = id;
  }
}
